script/lint_darwin_plugins.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
```

Comment:
Is there a reason this script can't be Dart? It seems like anything non-trivial should be Dart if it all possible, as it's easier to maintain.

Reply:
All the Cirrus tests are run from shell scripts, and this was closely cribbed from [check_publish.sh](https://github.com/flutter/plugins/blob/master/script/check_publish.sh).  I will defer to the plugins team as to whatever they find most maintainable (I'd prefer to not to be the sole owner of this effort).

Reply:
I'm fine deferring final decision to the plugins team. I would strongly encourage them to view adding a new non-trivial script as a good opportunity to begin a transition to Dart though :)

(Also worth considering is that at some point we're going to have Windows plugins in this repo, and probably will want to run scripts for them, and I cannot stress enough how much we want to those scripts to be in Dart rather than Batch.)

Reply:
So it's kind of scattered. A lot of our CI scripts start off in bash here, but then ultimately end up deferred to a completely separate Dart tool maintained in the [flutter/plugin_tools](https://github.com/flutter/plugin_tools) repo. It's kind of hard to follow but the flow is that CI calls a bash script, and the bash script downloads and runs random commands from that repo by calling the package via `flutter_plugin_tools`, see [these lines](https://github.com/flutter/plugins/blob/master/script/incremental_build.sh#L26-L28). 

I think it probably makes the most sense for this functionality to live as a command in that tool instead of in bash given how much is going on here. The meat of this is easier to do in shell since it needs to run `pod lint`, but I'm pretty sure that repo does expose process run commands that could be called instead. That said the bash is relatively straightforward here so I wouldn't block the PR on moving it either, I don't think maintaining it in this form would be a huge burden.

Reply:
I created https://github.com/flutter/flutter/issues/41511 and assigned it to myself.  I'll work on it once I convert a few more plugins to modules and get the plugin template done in flutter tools.